libmamba/src/core/package_cache.cpp

```diff
@@ -168,7 +169,26 @@ namespace mamba
             }
             else
             {
-                LOG_WARNING << "Package tarball '" << tarball_path.string() << "' is invalid";
+                std::stringstream msg;
```

Comment:
Consider putting all this in a separate function for clarity. Not a big need, I just suspect it might help reading. If you end up feeling it doesnt, ignore this comment.

Reply:
I think that inlining the message is clearer if it is only used once.